router.go

```diff
@@ -39,7 +39,7 @@ type Router struct {
 	contact         oaContact
 	servers         []oaServer
 	securitySchemes map[string]oaSecurityScheme
-	security        map[string][]string
+	security        []map[string][]string
```

Comment:
The spec shows "empty object" as a valid value--do we need to be able to express that? If so, does a nil map work here? Is this already handled by the nil handling in the Router.SecurityRequirement(...)?

Reply:
That's an uncommon case of "no auth" (or auth optional) which I was deliberately avoiding for now. We should be able to support it additively in the future if needed via another method on `Router`.